src/App.js

```diff
@@ -47,22 +86,54 @@ const App = () => {
     //    3 squares in each column match
     // 3. Go across each diagonal to see if 
     //    all three squares have the same value.
+    
+   let winningMoveList=[
+      [board[0][0],board[0][1],board[0][2]],
+      [board[1][0],board[1][1],board[1][2]],
+      [board[2][0],board[2][1],board[2][2]],
+      [board[0][0],board[1][0],board[2][0]],
+      [board[0][1],board[1][1],board[2][1]],
+      [board[0][2],board[1][2],board[2][2]],
+      [board[0][0],board[1][1],board[2][2]],
+      [board[0][2],board[1][1],board[2][0]],
 
+    ]
+
+
+
+    for(let i=0; i<winningMoveList.length;i++){
+    
+      let isWinner=false
+      if (winningMoveList[i][0].value === winningMoveList[i][1].value && winningMoveList[i][0].value ===winningMoveList[i][2].value  ){
+
+        if (winningMoveList[i][0].value === PLAYER_1) {
+          isWinner=true
+          setWinner('x')
+
+        }else if(winningMoveList[i][0].value === PLAYER_2){
+          isWinner=true
+          setWinner('o')
+  
+        }
+      }
+    }
   }
 
   const resetGame = () => {
     // Complete in Wave 4
+    setSquares(generateSquares());
+    setWinner(null);
   }
-
+console.log('winner',winner)
   return (
     <div className="App">
       <header className="App-header">
         <h1>React Tic Tac Toe</h1>
-        <h2>The winner is ... -- Fill in for wave 3 </h2>
-        <button>Reset Game</button>
+        <h2>Winner is {winner}</h2>
```

Comment:
Here is a ternary operator you can use to display the current player if the winner is null ``{winner === null ? `Current Player ${ currentPlayer }` : `Winner is ${ winner }`}``